src/julia.h

```diff
@@ -2337,24 +2337,32 @@ extern void (*real_siglongjmp)(jmp_buf _Buf, int _Value);
 
 #ifdef __clang_gcanalyzer__
 
+extern int had_exception;
 // This is hard. Ideally we'd teach the static analyzer about the extra control
 // flow edges. But for now, just hide this as best we can
-extern int had_exception;
-#define JL_TRY if (1)
-#define JL_CATCH if (had_exception)
+#define JL_TRY                                                    \
+    int i__try, i__catch; jl_handler_t __eh;                          \
+    size_t __excstack_state = jl_excstack_state();                \
+    jl_enter_handler(&__eh);                                      \
+    for (i__try=2; i__try; i__try--)                                 \
+        if (i__try==2)                                             \
```

Comment:
I don't think this models that we can enter the catch block from part-way through the try block, so it seems like this is still unsound?